src/test/java/plugins/GitPluginTest.java

```diff
@@ -203,16 +216,70 @@ public void update_submodules_recursively() throws IOException, InterruptedExcep
         job.startBuild().shouldSucceed();
     }
 
+
+    @Test
+    public void check_scm_changes() throws IOException, InterruptedException, SftpException, JSchException {
+        String filename = "test-file";
+        String content = "test content";
+        GitRepo repo = new GitRepo();
+
+        repo.add_file_and_commit("Test commit", filename);
+        repo.append_string_to_file(filename, content);
+        repo.add_file_and_commit("Test: changed content", filename);
+        repo.transferToDockerContainer(host, port);
+
+
+        job.useScm(GitScm.class)
+                .url(repoUrl)
+                .credentials(USERNAME)
+                .localDir("local_dir");
+        job.addShellStep("cd local_dir && test -f " + filename);
+        job.addShellStep("cd local_dir && echo '" + content + "' >> " + filename);
+        job.addShellStep("cd local_dir && nr=\"$(grep -c \""+ content + "\" " + filename + ")\" && if [ \"$nr\" = \"1\" ]; then exit 0; else exit 1; fi;");
+        job.save();
+
+
+        job.startBuild().shouldSucceed();
+    }
+
+    @Test
+    public void failed_check_scm_changes() throws IOException, InterruptedException, SftpException, JSchException {
```

Comment:
This test case makes no sense to me: it is basically the same as the last but does check for something that could never happen.